sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -821,18 +822,129 @@ def test_partial_dependence_unfitted(estimator):
 
 
 @pytest.mark.parametrize(
-    "Estimator, data",
+    "estimator, data",
     [
         (LinearRegression, multioutput_regression_data),
         (LogisticRegression, binary_classification_data),
     ],
 )
-def test_kind_average_and_average_of_individual(Estimator, data):
-    est = Estimator()
+def test_kind_average_and_average_of_individual(estimator, data):
+    est = estimator()
     (X, y), n_targets = data
     est.fit(X, y)
 
     pdp_avg = partial_dependence(est, X=X, features=[1, 2], kind="average")
     pdp_ind = partial_dependence(est, X=X, features=[1, 2], kind="individual")
     avg_ind = np.mean(pdp_ind["individual"], axis=1)
     assert_allclose(avg_ind, pdp_avg["average"])
+
+
+@pytest.mark.parametrize(
+    "estimator, data",
+    [
+        (LinearRegression, multioutput_regression_data),
+        (LogisticRegression, binary_classification_data),
+    ],
+)
+def test_partial_dependence_kind_individual_ignores_sample_weight(estimator, data):
+    est = estimator()
+    (X, y), n_targets = data
+    sample_weight = [1] + [0] * (len(X) - 1)
+    est.fit(X, y)
+
+    pdp_nsw = partial_dependence(est, X=X, features=[1, 2], kind="individual")
+    pdp_sw = partial_dependence(
+        est, X=X, features=[1, 2], kind="individual", sample_weight=sample_weight
+    )
+    assert_allclose(pdp_nsw["individual"], pdp_sw["individual"])
+    assert_allclose(pdp_nsw["values"], pdp_sw["values"])
+
+
+@pytest.mark.parametrize(
+    "estimator",
+    [
+        LinearRegression(),
+        LogisticRegression(),
+        RandomForestRegressor(),
+        GradientBoostingClassifier(),
+    ],
+)
+def test_partial_dependence_sample_weight_ind_equals_one(estimator):
+    X, y = iris.data, iris.target
+    preprocessor = make_column_transformer(
+        (StandardScaler(), [0, 2]), (RobustScaler(), [1, 3])
+    )
+    pipe = make_pipeline(preprocessor, estimator).fit(X, y)
+
+    for sample_weight_ind_equals_one in [0, 1, -1]:
```

Comment:
I think that we could parametrize this one outside. A more concise name would be `non_null_weight_idx`